sources/packages/backend/apps/api/src/route-controllers/report/_tests_/e2e/report.institutions.controller.exportReport.e2e-spec.ts

```diff
@@ -0,0 +1,283 @@
+import {
+  BEARER_AUTH_TYPE,
+  InstitutionTokenTypes,
+  authorizeUserTokenForLocation,
+  createTestingAppModule,
+  getAuthRelatedEntities,
+  getInstitutionToken,
+} from "../../../../testHelpers";
+import { parse } from "papaparse";
+import * as request from "supertest";
+import { SystemUsersService } from "@sims/services";
+import { AppInstitutionsModule } from "../../../../app.institutions.module";
+import { FormNames, FormService } from "../../../../services";
+import { TestingModule } from "@nestjs/testing";
+
+describe("ReportInstitutionsController(e2e)-exportReport", () => {
+  let app: INestApplication;
+  let appModule: TestingModule;
+  let db: E2EDataSources;
+  let collegeF: Institution;
+  let collegeFLocation: InstitutionLocation;
+  let collegeC: Institution;
+  let collegeCLocation: InstitutionLocation;
+  let systemUsersService: SystemUsersService;
+  let formService: FormService;
+  let programYear: ProgramYear;
+  const PROGRAM_YEAR_PREFIX = 2010;
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appModule = module;
+    db = createE2EDataSources(dataSource);
+    // Mock the form service to validate the dry-run submission result.
+    formService = await getProviderInstanceForModule(
+      appModule,
+      AppInstitutionsModule,
+      FormService,
+    );
+    // College F.
+    const { institution: collegeF } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeFLocation = createFakeInstitutionLocation({ institution: collegeF });
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+    // College C.
+    const { institution: collegeC } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeCLocation = createFakeInstitutionLocation({ institution: collegeC });
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+    systemUsersService = nestApplication.get(SystemUsersService);
+    // Program Year for the following tests.
+    programYear = await ensureProgramYearExists(db, PROGRAM_YEAR_PREFIX);
+  });
+
+  it("Should generate the offering details report when a report generation request is made with the appropriate program year and offering intensity.", async () => {
+    // Arrange
+
+    // Created 3 offerings as follows:
+    // 1st offering with 0 applications.
+    // 2nd offering with two applications for two different students belonging to the location of the institution for which the report is generated.
+    // 3rd offering with an application for a student belonging to the location of a different institution for which this report is not generated.
+    const auditUser = systemUsersService.systemUser;
```

Comment:
It does not seem right to use the system user to create programs.
Can we use the institution one?

Reply:
The `getAuthRelatedEntities` also can return the institution user authenticated. As shown below.
```ts
const { institution: collegeC, user: institutionUser } =
  await getAuthRelatedEntities(
    db.dataSource,
    InstitutionTokenTypes.CollegeCUser,
  );
```

Reply:
yes